lib/src/runner/configuration.dart

```diff
@@ -386,6 +386,31 @@ class Configuration {
     if (this == Configuration.empty) return other;
     if (other == Configuration.empty) return this;
 
+    List<String> foldTraceOnly = other._foldTraceOnly ?? _foldTraceOnly;
+    if (other._foldTraceOnly != null && _foldTraceExcept != null) {
+      foldTraceOnly = new Set.from(other._foldTraceExcept)
+          .difference(new Set.from(_foldTraceExcept))
+          .toList();
+    }
+
+    if (_foldTraceOnly != null && other._foldTraceExcept != null) {
+      foldTraceOnly = new Set.from(_foldTraceExcept)
+          .difference(new Set.from(other._foldTraceExcept))
+          .toList();
+    }
+    if (other._foldTraceOnly != null && _foldTraceOnly != null) {
+      foldTraceOnly = new Set.from(other._foldTraceOnly)
+          .intersection(new Set.from(_foldTraceOnly))
+          .toList();
+    }
+
+    List<String> foldTraceExcept = other._foldTraceExcept ?? _foldTraceExcept;
+    if (other._foldTraceExcept != null && _foldTraceExcept != null) {
+      foldTraceExcept = new Set.from(other._foldTraceExcept)
+          .union(new Set.from(_foldTraceExcept))
+          .toList();
+    }
```

Comment:
If you can assume that only one of `only` and `except` is null, you can clean up all these conditionals:

```dart
if (_foldTraceOnly != null) {
  if (other._foldTraceExcept != null) {
    // difference
  } else {
    // intersection
  }
} else {
  if (other._foldTraceExcept != null) {
    // union
  } else {
    // difference
  }
}